classes/image_types_ostree.bbclass

```diff
@@ -16,25 +9,20 @@ OSTREE_KERNEL ??= "${KERNEL_IMAGETYPE}"
 OSTREE_COMMIT_SUBJECT ??= "Commit-id: ${IMAGE_NAME}"
 OSTREE_COMMIT_BODY ??= ""
 OSTREE_UPDATE_SUMMARY ??= "0"
+OSTREE_TAR_SUFFIX ??= ".ostree.tar.bz2"
 
-export SYSTEMD_USED = "${@oe.utils.ifelse(d.getVar('VIRTUAL-RUNTIME_init_manager', True) == 'systemd', 'true', '')}"
+export OSTREE_ROOTFS ?= "${WORKDIR}/ostree-rootfs"
 
-IMAGE_CMD_ostree () {
-    if [ -z "$OSTREE_REPO" ]; then
-        bbfatal "OSTREE_REPO should be set in your local.conf"
-    fi
-
-    if [ -z "$OSTREE_BRANCHNAME" ]; then
-        bbfatal "OSTREE_BRANCHNAME should be set in your local.conf"
-    fi
+export SYSTEMD_USED = "${@oe.utils.ifelse(d.getVar('VIRTUAL-RUNTIME_init_manager', True) == 'systemd', 'true', '')}"
```

Comment:
I guess it's needed to be exported so far since it's being referred currently in IMAGE_CMD_ostree task as a shell variable, as follows:
...
if [ -n "$SYSTEMD_USED" ]; then
...

if we intend to remove the export syntax from it, then we need change the reference to be a bitbake variable expanding syntax as ${SYSTEMD_USED}.

I guess the author might has his reason to export it, but I can not tell by myself.

//Ming Liu

Reply:
Hi, Stefan:

You probably asked if the OSTREE_ROOTFS variable needs to be exported or not. I probably mis-understood your meaning.

If so, I agreed, it does not have to be exported, will fix it in V2.

Reply:
> However, I don't think that any subprocess or script uses `SYSTEMD_USED`... So I think we do not need to export that.

We used to use export a lot more than necessary. I'm not sure why, to be honest. It is quite likely that this particular example is also not necessary to export.

Reply:
I am on board with Agners, it looks like not need to be exported, I have just send a V3 including a patch to remove it, please kindly help me review.